### PR TITLE
[W4.3d][W15-A1]Poh Jie

### PR DESCRIPTION
**Pull request by pohjie**
Not my final product. There are some issues I have to check with my TA.

### CODE REVIEW

**Comment by pohjie** [comment]
Work in Progress

**Comment by pohjie** [comment]
Ready for review.

File: doc/UserGuide.md [hunk 1]
```diff
@@ -44,6 +44,7 @@ Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [t/TAG]...`
 > be seen using the `viewall` command.
 > 
 > Persons can have any number of tags (including 0)
+> Address will be split into 4 subclasses- block, street, unit, postal code
```
Comment (okkhoy) [inline doc/UserGuide.md]: :+1:

File: src/seedu/addressbook/commands/AddCommand.java [hunk 2]
```diff
@@ -4,13 +4,7 @@
...
 
 import seedu.addressbook.data.exception.IllegalValueException;
-import seedu.addressbook.data.person.Address;
-import seedu.addressbook.data.person.Email;
-import seedu.addressbook.data.person.Name;
-import seedu.addressbook.data.person.Person;
-import seedu.addressbook.data.person.Phone;
-import seedu.addressbook.data.person.ReadOnlyPerson;
-import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.person.*;
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/AddCommand.java]: Don't import everything via `*`. Import the classes individually.

File: src/seedu/addressbook/commands/AddCommand.java [hunk 3]
```diff
@@ -46,11 +40,15 @@ public AddCommand(String name,
         for (String tagName : tags) {
             tagSet.add(new Tag(tagName));
         }
+        Address superclassAddress = new Address(address, isAddressPrivate);
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/AddCommand.java]: any specific reason for the variable name `superclassAddress`?

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -13,6 +16,10 @@
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
     public final String value;
+    public Block block;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: - It is not necessary to retain the `value` member. Think what can go wrong maintaining duplicate data?
- Don't make the members `public` by default. Always start with the most restrictive access modifier. If that is not sufficient, then upgrade step by step.

File: src/seedu/addressbook/data/person/Address.java [hunk 5]
```diff
@@ -27,6 +34,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+
+        String[] addressSubClass = trimmedAddress.split("\\s*,\\s*");
+        this.block = new Block(addressSubClass[0], isPrivate);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: What does our coding standard say about using numbers in array indices? [Hint! read magic numbers]

File: src/seedu/addressbook/data/person/Block.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's address in the address book.
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: Not really. This represents just the block in the address.

File: src/seedu/addressbook/data/person/Block.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+ * Guarantees: immutable; is valid as declared in {@link #isValidBlock(String)}
+ */
+public class Block {
+
+    public static final String EXAMPLE = "123";
+    public static final String MESSAGE_BLOCK_CONSTRAINTS = "Person block can be in any format";
+    public static final String BLOCK_VALIDATION_REGEX = ".+";
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: Isn't it better to make the constraint more explicit and restrictive?

File: src/seedu/addressbook/data/person/Block.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        String trimmedBlock = block.trim();
+        this.isPrivate = isPrivate;
+        if (!isValidBlock(trimmedBlock)) {
+            throw new IllegalValueException(MESSAGE_BLOCK_CONSTRAINTS);
+        }
+        this.value = trimmedBlock;
+    }
+
+    /**
+     * Returns true if a given string is a valid person address.
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: copy paste error :-P this just validates the block!

File: src/seedu/addressbook/data/person/Unit.java [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+    public static final String EXAMPLE = "123";
+    public static final String MESSAGE_UNIT_CONSTRAINTS = "Person block can be in any format";
+    public static final String UNIT_VALIDATION_REGEX = ".+";
+
+    public final String value;
+    private boolean isPrivate;
+
+    /**
+     * Validates given block.
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Unit.java]: :-P again copy paste errors?

File: src/seedu/addressbook/storage/jaxb/AdaptedPerson.java [hunk 10]
```diff
@@ -1,3 +1,5 @@
+// Having issue resolving this java file. Also, where in the program does it process the a/ to split it up into subclass?
```
Comment (okkhoy) [inline src/seedu/addressbook/storage/jaxb/AdaptedPerson.java]: - Comments like this are not useful in the code. Hence should not be retained. You can ask questions/explain intention in Github comments like what I am doing now :-)

- To answer the question: DId you manage to find where the input string is split into email, phone, address? That could be a potential place to further split to parts. Second location would be in the address class itself. Think which implementation maintains good SLAP.

File: test/expected.txt [hunk 11]
```diff
@@ -73,58 +73,58 @@
...
-|| Enter command: || [Command entered:  add Valid Name p/not_numbers e/valid@e.mail a/valid, address]
+|| Enter command: || [Command entered:  add Valid Name p/not_numbers e/valid@e.mail a/valid, address, boo, yea]
 || Person phone numbers should only contain numbers
 || ===================================================
-|| Enter command: || [Command entered:  add Valid Name p/12345 e/notAnEmail a/valid, address]
+|| Enter command: || [Command entered:  add Valid Name p/12345 e/notAnEmail a/valid, address, boo, yea]
 || Person emails should be 2 alphanumeric/period strings separated by '@'
 || ===================================================
-|| Enter command: || [Command entered:  add Valid Name p/12345 e/valid@e.mail a/valid, address t/invalid_-[.tag]
+|| Enter command: || [Command entered:  add Valid Name p/12345 e/valid@e.mail a/valid, address, boo, yea t/invalid_-[.tag]
```
Comment (okkhoy) [inline test/expected.txt]: :+1: for updating the tests

**Comment by okkhoy** [comment]
@pohjie 
Some comments added. Please ack & close the PR after reading comments.

File: src/seedu/addressbook/storage/jaxb/AdaptedPerson.java (hunk 10 above)
Comment (pohjie) [reply src/seedu/addressbook/storage/jaxb/AdaptedPerson.java]: Yes I did, and I actually added the commit for myself. That being said, now I'm clearer on what are the expectations for a commit when others are going to view it. Thanks!

**Pull request closed without merging**